.github/workflows/linux.yml

```diff
@@ -0,0 +1,51 @@
+name: "Linux builds"
+on: push
+jobs:
+
+  build_ubuntu_gcc9_debug:
+    name: "GCC 9 debug (ubuntu-18.04)"
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - run: sudo ./scripts/build.sh --install-deps --compiler-version 9 --release debug
+      - name: "Summarize warnings"
+        run: python3 ./scripts/count-warnings.py build.log
+
+  build_ubuntu_gcc9_lto:
+    name: "GCC 9 small LTO (ubuntu-18.04)"
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - run: sudo ./scripts/build.sh --install-deps --compiler-version 9 --release small --lto
+      - name: "Summarize warnings"
+        run: python3 ./scripts/count-warnings.py build.log
+
+  build_ubuntu_gcc_defaults:
+    name: "GCC"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-16.04]
+    steps:
+      - uses: actions/checkout@v1
+      - run: sudo ./scripts/build.sh --install-deps
+      - name: "Summarize warnings"
+        run: python3 ./scripts/count-warnings.py build.log
+
+  build_ubuntu_clang_8_debug:
+    name: "Clang 8 debug (ubuntu-18.04)"
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - run: sudo ./scripts/build.sh --install-deps --compiler clang --compiler-version 8 --release debug
+      - name: "Summarize warnings"
+        run: python3 ./scripts/count-warnings.py build.log
+
+  build_ubuntu_clang_8_lto:
+    name: "Clang 8 small LTO (ubuntu-18.04)"
```

Comment:
LTO build is cool and important for release builds, but do we need it in CI on every push?

Reply:
If we have a nightly build, then that's probably the best place for LTO.    

We definitely want to exercise LTO in some capacity with both compilers because it has historically been the most error-prone and difficult task for both compilers - and it's still under significantly improvement across all aspects (speed, memory, and threading). 

If code is too "assumptive", the link step of LTO will fail for various reasons.  Gentoo for years has been pushing fixes to upstream packages that fail to build with LTO, and there's been a multi-year effort by Andi Kleen to make the Linux kernel build with LTO; and it still regresses in this regard - but everyone is trying to get it there. 

The more STL we use, the more LTO will catch as well (if we're missing explicit instantiations). 
So in this regard, using LTO will catch things early when they go off the rails.

Reply:
Knocked out the debug and LTO builds, in favor of using those in the nightly build & asset packager.